cluster-autoscaler/cloudprovider/gce/gce_url.go

```diff
@@ -83,15 +98,19 @@ func IsInstanceTemplateRegional(templateUrl string) (bool, error) {
 	return regexp.MatchString("(/projects/.*[A-Za-z0-9]+.*/regions/)", templateUrl)
 }
 
-func parseGceUrl(url, expectedResource string) (project string, zone string, name string, err error) {
-	reg := regexp.MustCompile(fmt.Sprintf("https://.*/projects/.*/zones/.*/%s/.*", expectedResource))
-	errMsg := fmt.Errorf("wrong url: expected format <url>/projects/<project-id>/zones/<zone>/%s/<name>, got %s", expectedResource, url)
+func parseGceUrl(prefix, url, expectedResource string) (project string, zone string, name string, err error) {
+	reg := regexp.MustCompile(fmt.Sprintf("%sprojects/.*/zones/.*/%s/.*", prefix, expectedResource))
+	if prefix != "" {
+		prefix = "<url>/"
+	}
+	errMsg := fmt.Errorf("wrong url: expected format %sprojects/<project-id>/zones/<zone>/%s/<name>, got %s", prefix, expectedResource, url)
 	if !reg.MatchString(url) {
 		return "", "", "", errMsg
 	}
-	splitted := strings.Split(strings.Split(url, projectsSubstring)[1], "/")
-	project = splitted[0]
-	zone = splitted[2]
-	name = splitted[4]
+
+	splitted := strings.Split(url, "/")
```

Comment:
What if we used capturing groups in the regular expression instead of splitting by slashes? 

So the regexp changes to "%sprojects/(.*)/zones/(.*)/%s/(.*)" and then use https://pkg.go.dev/regexp#Regexp.FindStringSubmatch?

Reply:
Done